app/main/messaging.py

```diff
@@ -0,0 +1,392 @@
+                          'account at {}'
+                          .format(url_for('account.register', _external=True)))
+        else:
+            twiml.message('See all your reports at {}'
+                          .format(url_for('reports.view_my_reports',
+                                          _external=True)))
+
+        if image_job_id is not None:
+            get_queue().enqueue(
+                attach_image_to_incident_report,
+                depends_on=image_job_id,
+                incident_report=new_incident,
+                image_job_id=image_job_id,
+            )
+
+        # reset report variables/cookies
+        step = 0
+
+    else:
+        twiml.message('Welcome to {}! Please reply "report" to report an '
+                      'idling incident.'
+                      .format(current_app.config['APP_NAME']))
+
+    response = make_response(str(twiml))
+
+    set_cookie(response, 'messagecount', str(step))
+    set_cookie(response, 'agency_name', agency_name)
+    set_cookie(response, 'vehicle_id', vehicle_id)
+    set_cookie(response, 'license_plate', license_plate)
+    set_cookie(response, 'duration', str(duration))
+    set_cookie(response, 'description', description)
+    set_cookie(response, 'location', location)
+    set_cookie(response, 'picture_url', picture_url)
+
+    return response
+
+
+def handle_create_report(agency_name, description, duration, license_plate,
+                         location, picture_url, vehicle_id, phone_number):
+    lat, lon = geocode(location)
+    agency = Agency.get_agency_by_name(agency_name)
+    if agency is None:
+        agency = Agency(name=agency_name, is_official=False,
+                        is_public=True)
+        db.session.add(agency)
+        db.session.commit()
+
+    new_incident = IncidentReport(
+        agency=agency,
+        vehicle_id=vehicle_id,
+        license_plate=license_plate if license_plate else None,
+        duration=timedelta(minutes=duration),
+        description=description,
+        location=Location(
+            latitude=lat,
+            longitude=lon,
+            original_user_text=location
+        ),
+        picture_url=picture_url if picture_url else None,
+        user=User.query.filter_by(phone_number=phone_number).first()
+    )
+    db.session.add(new_incident)
+    db.session.commit()
+
+    return new_incident
+
+
+def handle_start_report(step, twiml):
+    step = 1
+    twiml.message('What is your location? Be specific! (e.g. "34th and '
+                  'Spruce in Philadelphia PA")')
+    return step
+
+
+def handle_location_step(body, step, twiml):
+    validator_form = IncidentReportForm()
+    errors = data_errors(form=validator_form, field=validator_form.location,
+                         data=body)
+
+    if geocode(body) == (None, None):
+        errors.append('We could not find that location. Please respond with a '
+                      'full address including city and state.')
+    if len(errors) == 0:
```

Comment:
It would probably work the same but I think it's much better to be explicit here. 

See the debate [here](http://stackoverflow.com/a/7302987/3624201)